### ISSUE TITLE
DB Connection Exhaustion

### ISSUE THREAD

**Issue by starzengarters** [starzengarters/FIAB#1]
I occasionally encounter an error
```
Npgsql.PostgresException (0x80004005): 53300: remaining connection slots are reserved for roles with the SUPERUSER attribute
```

I confirmed I have 100 slots on the DB, so that means some page or component is making connections to the DB and not letting go. I'm missing a `using` or some sort of `IDisposable` magic isn't working somewhere.

**Comment by starzengarters** [comment]
Possibly corrected in ddf11182.

**Comment by starzengarters** [comment]
I think this did get solved when I ensured the `finally` blocks in `OnAfterRenderAsync` overrides only ran on `firstRender`.
